pkg/controller/middleware/i18n.go

```diff
@@ -35,6 +41,69 @@ var rightAlignLanguages = map[string]struct{}{
 	"ar": {},
 }
 
+// translationReloader maintains a pointer to the LocaleMap
+// and will manage refreshing translations from the database
+// on the configured interval.
+type translationReloader struct {
+	locales    *i18n.LocaleMap
+	db         *database.Database
+	cacher     cache.Cacher
+	lastUpdate time.Time
+	period     time.Duration
+	mu         sync.RWMutex
+}
+
+// reload will see if we are due for a refresh, otherwise exit quickly.
+func (tr *translationReloader) reload(ctx context.Context) {
+	// Check and see if we need to reload.
+	now := time.Now().UTC()
+	tr.mu.RLock()
+	if tr.lastUpdate.Add(tr.period).After(now) {
+		tr.mu.RUnlock()
+		return
+	}
+	tr.mu.RUnlock()
+
+	// Needs refresh.
+	tr.mu.Lock()
+	defer tr.mu.Unlock()
+
+	// This thread lost the race to actually do the refresh, shucks.
+	if tr.lastUpdate.Add(tr.period).After(now) {
+		return
+	}
+
+	logger := logging.FromContext(ctx)
+
+	// Read the translations.
+	translations, err := tr.db.ListDynamicTranslationsCached(ctx, tr.cacher)
+	if err != nil {
+		logger.Errorw("unable to read dynamic_translations", "error", err)
+	}
+
+	tr.locales.SetDynamicTranslations(translations)
+	tr.lastUpdate = now
+}
+
+func LoadDynamicTranslations(locales *i18n.LocaleMap, db *database.Database, cacher cache.Cacher, period time.Duration) mux.MiddlewareFunc {
+	state := &translationReloader{
+		locales: locales,
+		db:      db,
+		cacher:  cacher,
+		period:  period,
+		// default lastUpdate to time.Zero
+	}
+	ctx := context.Background()
+	state.reload(ctx)
```

Comment:
Shouldn't this return an error if it fails?

Reply:
made it so it returns an error the first time (Will block startup), but then won't fail on subsequent runs